lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,174 @@
+                       bindEvents: false,
+                       titleIconName: 'fa-globe',
+                       options: this.countries,
+                       helpMessage: gettext("The country that team members primarily identify with.")
+                   });
+               },
+
+               render: function() {
+                   this.$el.html(_.template(edit_team_template)({}));
+                   this.set(this.teamNameField, '.team-required-fields');
+                   this.set(this.teamDescriptionField, '.team-required-fields');
+                   this.set(this.optionalDescriptionField, '.team-optional-fields');
+                   this.set(this.teamLanguageField, '.team-optional-fields');
+                   this.set(this.teamCountryField, '.team-optional-fields');
+                   return this;
+               },
+
+               set: function(view, selector) {
+                   this.$el.find(selector).append(view.$el);
+                   view.render();
+               },
+
+               createTeam: function () {
+                   var teamName = this.teamNameField.fieldValue();
+                   var teamDescription = this.teamDescriptionField.fieldValue();
+                   var teamLanguage = this.teamLanguageField.fieldValue();
+                   var teamCountry = this.teamCountryField.fieldValue();
+
+                   var validation = this.validateTeamData(teamName, teamDescription);
+                   if (validation.status === false) {
+                       this.showMessage(validation.message);
+                       return;
+                   }
+
+                   var data = {
+                       course_id: this.courseId,
+                       topic_id: this.topicId,
+                       name: teamName,
+                       description: teamDescription,
+                       language: _.isNull(teamLanguage) ? '' : teamLanguage,
+                       country: _.isNull(teamCountry) ? '' : teamCountry
+                   };
+
+                   // Send AJAX request to Teams API
+                   var view = this;
+                   $.ajax({
+                       type: 'POST',
+                       url: this.teamsUrl,
+                       data: data
+                   }).done(function () {
+                       Backbone.history.loadUrl(Backbone.history.fragment);
+                   }).fail(function (jqXHR) {
+                       view.showMessage(gettext('An error occurred. Please try again.'));
+                   });
+               },
+
+               validateTeamData: function (teamName, teamDescription) {
+                   var status = true,
+                       message = '',
+                       messageMissing = gettext("Your team could not be created because some required information is missing."),
+                       messageIncorrect = gettext("Your team could not be created because some required information is incorrect.");
```

Comment:
@muhammad-ammar Thanks for clarifying. Your suggestion is good with the exception of removing the word "required"; "highlighted" is sufficient:
"Your team could not be created. Check the highlighted fields below and try again."
Thanks!